src/service/gtfs-flex-service.ts

```diff
@@ -50,35 +36,51 @@ class GtfsFlexService implements IGtfsFlexService {
     }
 
     async getGtfsFlexById(id: string): Promise<FileEntity> {
-        // get a gtfsFlex repository to perform operations with gtfsFlex
-        const gtfsFlexRepository = AppDataSource.getRepository(FlexVersions);
+        const query = {
+            text: 'Select file_upload_path from flex_versions WHERE tdei_record_id = $1',
+            values: [id],
+        }
 
-        // load a gtfsFlex by a given gtfsFlex id
-        const gtfsFlex: FlexVersions | any = await gtfsFlexRepository.findOneBy(
-            {
-                tdei_record_id: id
-            });
+        let result = await dbClient.query(query);
+
+        if (result.rows.length == 0) throw new HttpException(400, "Record not found");
 
         const storageClient = Core.getStorageClient();
         if (storageClient == null) throw console.error("Storage not configured");
-        let url: string = decodeURIComponent(gtfsFlex?.file_upload_path);
+        let url: string = decodeURIComponent(result.rows[0].file_upload_path);
         return storageClient.getFileFromUrl(url);
     }
 
     async createAGtfsFlex(flexInfo: FlexVersions): Promise<GtfsFlexDTO> {
         try {
-            // get a gtfsFlex repository to perform operations with gtfsFlex
-            const gtfsFlexRepository = AppDataSource.getRepository(FlexVersions);
             flexInfo.file_upload_path = decodeURIComponent(flexInfo.file_upload_path!);
-            // create a real gtfsFlex object from gtfsFlex json object sent over http
-            const newGtfsFlex = gtfsFlexRepository.create(flexInfo);
+            const queryObject = {
+                text: `INSERT INTO public.flex_versions(tdei_record_id, 
+                    confidence_level, 
+                    tdei_org_id, 
+                    tdei_service_id, 
+                    file_upload_path, 
+                    uploaded_by,
+                    collected_by, 
+                    collection_date, 
+                    collection_method, valid_from, valid_to, data_source,
+                    flex_schema_version)
+                    VALUES ($1,0,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12)`.replace(/\n/g, ""),
+                values: [flexInfo.tdei_record_id, flexInfo.tdei_org_id, flexInfo.tdei_service_id, flexInfo.file_upload_path, flexInfo.uploaded_by
+                    , flexInfo.collected_by, flexInfo.collection_date, flexInfo.collection_method, flexInfo.valid_from, flexInfo.valid_to, flexInfo.data_source, flexInfo.flex_schema_version],
```

Comment:
better to keep this query as a member function of `FlexVersions` object itself.

Reply:
Taken care